packages/vscode-extension/src/debugging/DebugAdapter.ts

```diff
@@ -203,9 +203,13 @@ export class DebugAdapter extends DebugSession {
     // some logs may baypass that, especially when printed in initialization phase, so we
     // need to detect whether the wrapper has added the stack info or not
     // We check if there are more than 3 arguments, and if the last one is a number
+    // We also check if the log is not internal to avoid exposing it as part of
+    // application logs.
     const argsLen = message.params.args.length;
     let output: OutputEvent;
-    if (argsLen > 3 && message.params.args[argsLen - 1].type === "number") {
+    if (message.params.args[0].value === "__RNIDE_INTERNAL") {
+      return;
```

Comment:
add comment here to explain why we don't do anything with those logs

Reply:
isn't this in this explained by this comment? 
    
    // We filter out logs that start with __RNIDE_INTERNAL as those are messages
    // used by IDE for tracking the app state and should not appear in the VSCode
    // console.

Reply:
if the log doesn't appear anywhere what's its purpose?

Reply:
it appears in metro log feed with type: `client_log`, can be useful for us to debug when runtime started.

Reply:
yes, that's exacly what you should write here to explain why we just return and do nothing with the log.